.github/workflows/insights-code-analysis.yml

```diff
@@ -0,0 +1,188 @@
+name: 'insights static code analysis'
+# Runs static code analysis and linting on changed files.
+#
+on: 
+  pull_request:
+    paths:
+      - insights/**
+
+env:
+  # terraform versions
+  tf_version: '0.12.x'
+
+jobs:
+  ## CHANGED FILES
+  # searches changed files and add outputs for use by other jobs
+  # e.g., ${{ needs.changed_files.outputs.tf }} would output all changed terraform (*.tf) files
+  changed_files:
+    name: checking
+    runs-on: ubuntu-latest
+    outputs:
+      all: ${{ steps.changed_files.outputs.all }}
+      sh: ${{ steps.changed_files.outputs.sh }}
+      tf: ${{ steps.changed_files.outputs.tf }}
```

Comment:
Is this worth the extra complexity? Our TF jobs are pretty fast, and I think this part may be overkill, compared to the `paths` qualifier above. (If that works how I think it works.)

Reply:
It allows us to plug in other linters as needed. And it can speed things up (shell checks can take a while). Was trying to future proof.

Reply:
Also, I needed a way to find all the shell scripts instead of relying on file extensions. This method allows me to find all shell scripts, `foo` and `foo.sh` as long as they have shebangs.